### ISSUE TITLE
Under the images screen, I would like to have a update button, which would check for updates of an image.

### ISSUE THREAD

**Issue by rhatdan** [containers/podman-desktop#923]
### Is your enhancement related to a problem? Please describe

Would be nice to be able to easily update one or more all images in your store.

### Describe the solution you'd like

Add updat image (Greyed out if none selected).  If I select then do a podman pull of all selected images to see if their are updates available.

**Comment by axel7083** [comment]
I was looking at the backlog and saw this was in the sprint, but probably need some clarification:

## Introduction

We may have a lot of images in the store, so at first sight I would discourage allowing user to select all and click on update as this could be... terrific.

## Inspiration 

Reading the issue I though about my IDE, and how it display the branch statuses I have locally. Here is an example

![Image](https://github.com/user-attachments/assets/86a381a1-f499-4c26-8ff7-617b96e45057)

In the example above, the `fork/jeffmaury/GH-9124` has been rebased, so I am not sync with it, but I could git push force to restore it to my local one.

The `fix/status-bar-progress` has new commits that I do not have locally, but I am still based on the same than the remote.

To be able to update the statuses of the branches, I can click on fetch

![Image](https://github.com/user-attachments/assets/391a622f-de8b-43a9-a1f7-2a5375b67684)

This is something quick in theory, but when you have dozens of remote configured it can take up to a minute.

For every image we have locally, we know if the tag is a sha256 or not, and we know the registry api can tell us if we are updated or not.

## Solution proposal

We could for example have a `Fetch Update` button on the top right, and Podman Desktop could filter images that are linked to a registry (not localhost), and image with a tag without sha256, to check for update.

Here is a mock-up of how we could display it the information that the image is not in sync with the registry

![Image](https://github.com/user-attachments/assets/32286dfa-9853-4e6d-bdb2-85b4036102c6)

Then, we could inside the image's actions add a new operation to update it.

![Image](https://github.com/user-attachments/assets/010ceebb-11da-43e0-9033-389397297307)

## Performance consideration

Fetching should be a user request task, as this would induce a lot of calls to the registries apis. We should also limit the maximum to avoid

**Comment by benoitf** [comment]
I think the first step is a button update that only pull the same image name/tag (and if remote tag is not existing then of course it'll say there is an error)


> We could for example have a Fetch Update button on the top right, and Podman Desktop could filter images that are linked to a registry (not localhost), and image with a tag without sha256, to check for update.

I can named an image `docker.io/httpd:florent` and it's not a tag available on the remote (even if I used a good prefix)

First step = only update the image if clicked

Later improvement: periodic checks / status indicator, etc.

**Comment by axel7083** [comment]
> I can named an image docker.io/httpd:florent and it's not a tag available on the remote (even if I used a good prefix)

In that case if the tag is not available, we could have a push icon (ideas, not in the first step, need discussion etc.)

![Image](https://github.com/user-attachments/assets/798baec8-0653-4125-930f-4d7e94223209)

**Comment by axel7083** [comment]
> I think the first step is a button update that only pull the same image name/tag (and if remote tag is not existing then of course it'll say there is an error)

I think we should reuse the `Pull Image` page to do the update, as error handling etc. is something complex. Here is a quick POC, where I added an `Update Image` action to images that redirect to the `Pull Image` page with prefilled input.

https://github.com/user-attachments/assets/d6383f80-1daa-4649-8011-4db5b675c56f

> ⚠ this require change in a few component, first the `Typehead` component do not take default value, so this is a problem; the PullImage need a mechanism to received image to pull, here I quickly use the query param and use `router.goto('images/pull?imageName=...')`

I will mark this issue as `postponed` and add it to the next ` Bi-Weekly Team meeting` so we can discuss this in more depth.

Here is the POC branch for reference https://github.com/axel7083/podman-desktop/tree/poc/image-update-mechanism

**Comment by Firewall** [comment]
> I will mark this issue as postponed and add it to the next  Bi-Weekly Team meeting so we can discuss this in more depth.
> 
> Here is the POC branch for reference [axel7083/podman-desktop@poc/image-update-mechanism](https://github.com/axel7083/podman-desktop/tree/poc/image-update-mechanism?rgh-link-date=2024-11-08T13%3A41%3A56.000Z)

@axel7083 Do you know the result of the bi-weekly meeting discussion? What is required to move this back in `in progress`?

**Comment by axel7083** [comment]
> @axel7083 Do you know the result of the bi-weekly meeting discussion? What is required to move this back in `in progress`?

Hummm, I don't remind this being included in the bi-weekly recently?

**Comment by bmahabirbu** [comment]
Hi all threw this up for the ux call

I have an implementation up but would like a green light for the ui

**Comment by bmahabirbu** [comment]
Split the pr into multiple parts since theres a bit of friction on the ui side

**Comment by jiridostal** [comment]
@bmahabirbu hello, this has been in many past sprints with On Hold status, can we move it out or unblock this issue?

**Comment by bmahabirbu** [comment]
Working on this now see https://github.com/podman-desktop/podman-desktop/pull/18003 for the final pr for this feature
